spec/rubocop/target_ruby_spec.rb

```diff
@@ -42,39 +42,39 @@
           content = <<~HEREDOC
             Gem::Specification.new do |s|
               s.name = 'test'
-              s.required_ruby_version = '>= 2.7.2'
+              s.required_ruby_version = '>= 3.2.2'
```

Comment:
As alluded to in the PR description, most of these changes to Ruby version numbers in this PR are not necessary. The specs continue to pass without these changes. The reason that I am making these changes is because these specs were at risk of passing, even when the relevant functionality has been broken, because the specs previously expected `2.7` to be returned, which also happens to be the current default Ruby version. By changing the specs to expect a non-default Ruby version, they will now fail, if/when the relevant functionality is broken (as, indeed, I did break it, in one of my initial attempts at fixing this bug).